services/validatorWorker/heartbeat.js

```diff
@@ -0,0 +1,22 @@
+const assert = require('assert')
+const { persistAndPropagate } = require("./lib/propagation")
+
+function heartbeat(adapter, channel){
+	const whoami = adapter.whoami()
+	const validatorIdx = channel.validators.indexOf(whoami)
+	assert.ok(validatorIdx !== -1, 'validatorTick: sending heartbeat for a channel where we are not validating')
+	const otherValidators = channel.spec.validators.filter(v => v.id != whoami)
+
+	const timestamp = `${new Date().getTime() / 1000}`
+
+	return adapter.sign(timestamp)
```

Comment:
I think you should hash the timestamp so we always give the same format to `adapter.sign` 
also this needs to include the channelId too, as it's vulnerable to replay attacks too

perhaps we can use the new function that we added to the adapter as well? getSignableStateRoot

Reply:
done